### ISSUE TITLE
Route53 + CNAME wildcard record leads to wrong Hosted Zone being chosen

### ISSUE THREAD

**Issue by werbitzky** [xenolf/lego#248]
I use rancher-letsencrypt (https://github.com/janeczku/rancher-letsencrypt) of which lego is an upstream library.

Lego seems to elect the wrong Hosted Zone from Route53 in the following scenario:

Two **Hosted Zones**:

* a.foo.com
* b.foo.com

b.foo.com has a wildcard CNAME record:

_*.b.foo.com_ points to _lb.a.foo.com_

I try to get certificates for _svc.b.foo.com_. Now Lego apparently tries to add a record for _\_acme-challenge.svc.**b**.foo.com_ in the Record Zone _**a**.foo.com_ which of course is rejected by Route53:

```
Failed to change Route 53 record set: InvalidChangeBatch: RRSet with DNS name _acme-challenge.svc.b.foo.com. is not permitted in zone a.foo.com.\n\tstatus code: 400
```

If I remove the wildcard _CNAME_ the right Record Zone is chosen.

@janeczku pointed out that it might be a bug with ```acme.FindZoneByFqdn()```

https://github.com/janeczku/rancher-letsencrypt/issues/23

**Comment by xenolf** [comment]
Thanks for submitting this issue. We had a few issues with `acme.FindZoneByFqdn()` in the last month. I'll have a look at how I can improve it.

**Comment by janeczku** [comment]
So the problem is that the recursive name server is returning the SOA for the zone to which the CNAME is pointing to in the authority section of the reply.
See: `dig SOA mail.google.com`

In order to determine the zone correctly for CNAME'd domains we need to recurse up the labels of the domain name until we get the SOA record in the answer section.

**Comment by werbitzky** [comment]
thanks a lot @janeczku! this one looks good for a fix

**Comment by xenolf** [comment]
Closing this with ref to the merged PR #254
